crypto/src/memo.rs

```diff
@@ -75,7 +82,17 @@ impl MemoPlaintext {
             .try_into()
             .map_err(|_| anyhow!("could not fit plaintext into memo size"))?;
 
-        Ok(MemoPlaintext(plaintext_bytes))
+        // Strip zero-byte padding
+        let no_pad: Vec<u8> = MemoPlaintext::strip_padding(plaintext_bytes);
+        let plaintext_str = String::from_utf8_lossy(no_pad.as_slice());
+        Ok(MemoPlaintext(plaintext_str.to_string()))
+    }
+
+    /// Removes zero-byte padding from a slice of bytes
+    pub fn strip_padding(input: [u8; MEMO_LEN_BYTES]) -> Vec<u8> {
+        // TODO: Make this less naive; should only strip trailing contiguous zero-bytes.
+        let result: Vec<u8> = input.into_iter().filter(|&x| x != 0u8).collect();
+        result
```

Comment:
`trim_end_matches` looks like it might be useful here? https://doc.rust-lang.org/stable/std/primitive.str.html#method.trim_end_matches